packages/kovan-faucets/src/ts/handler.ts

```diff
@@ -76,7 +75,6 @@ export class Handler {
     // tslint:disable-next-line:prefer-function-over-method
     private _createProviderEngine(rpcUrl: string) {
         const engine = new ProviderEngine();
-        engine.addProvider(new NonceSubprovider());
```

Comment:
I don't think this is the right solution to this issue. The reason we added the nonce subprovider was because we are submitting multiple txs to the network before waiting for them to confirm. Asking the network for the next nonce would return the same nonce until the txs confirm whereas this subprovider updates the nonce immediately. Have we tried this "fix" when requesting ETH multiple times in quick succession?

Reply:
I think we need to do more investigating into why exactly the nonceTracker can end up with the wrong nonce in the first place. Is there really no way to force-clear the nonce tracker if it gets into an inconsistent state?

Reply:
Yep there is more information in #333 including what happens when we attempt multiple requests in quick succession, we will get a number of errors from nonce re-use:

I.e if someone spams to the same account:

```
Unexpected err: [object Object] - {"code":-32010,"message":"Transaction with the same hash was already imported."}
```

It's difficult to reproduce the Infura Nonce value being different to what is acceptable when we sendTransaction. We are getting stuck with a cached nonce that is not valid on the next sendTransaction and it is the NonceSubprovider caching this incorrect value indefinitely. Is the test faucet really under that high of a load for the nonce optimisation?

If so, I think we either clone the Noncetracker and add it to our subproviders, maintaining the same functionality but clearing any Cached values on error, or try and get this fixed upstream. Discussion started [here](https://github.com/MetaMask/provider-engine/issues/213).

Reply:
So, you only need to submit two requests to the faucet in less then 15 seconds in order to cause this issue to reproduce. This is not **heavy** load.

The issue here is Infura being janky. I think we can fix the issue of getting an incorrect nonce value cached by not using Infura. If we want to keep using Infura then we should move NonceTracker to subproviders.

This is my hunch, but if I find some extra time I'll try debugging to make sure.

Reply:
For those following along, we're going to create a NonceTracker in the `subproviders` package. Allowing us to better handle error cases (in the occasional event of a faulty node) while continuing to reduce network usage of nonce requests.

I'll open this up in a separate PR.

Reply:
nope.